lib/src/frontend/expect.dart

```diff
@@ -11,6 +11,12 @@ import '../backend/invoker.dart';
 import '../utils.dart';
 import 'async_matcher.dart';
 
+/// A future that emits `null`.
+///
+/// We cache and re-use this value to avoid adding a new microtask hit for each
```

Comment:
have we confirmed that this solves the flutter scenario?

Another pattern we could consider is to return `null` from `_expect` and always guarantee a `Future` from `expectLater` with something like `=> _expect(...) ?? _emptyFuture;`